app/src/main/java/com/kickstarter/ui/activities/CommentsActivity.kt

```diff
@@ -27,10 +46,48 @@ class CommentsActivity :
         setContentView(view)
         binding.commentsRecyclerView.adapter = adapter
 
+        val loadMoreListView = PaginationHandler(
+            adapter,
+            binding.commentsRecyclerView,
+            binding.commentsSwipeRefreshLayout
+        )
+
+        loadMoreListView.onRefreshListener = {
+            viewModel.inputs.refresh()
+        }
+
+        loadMoreListView.onLoadMoreListener = {
+            viewModel.inputs.nextPage()
+        }
+
+        viewModel.outputs.enablePagination()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                loadMoreListView.loadMoreEnabled = it
+            }
+
+        viewModel.outputs.isLoadingMoreItems()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                loadMoreListView.isLoading(it)
+                binding.commentsLoadingIndicator.isVisible = it
+            }
+
+        viewModel.outputs.isRefreshing()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                loadMoreListView.refreshing(it)
+            }
+
         viewModel.outputs.commentsList()
             .compose(bindToLifecycle())
             .observeOn(AndroidSchedulers.mainThread())
-            .subscribe { it?.let { comments -> adapter.takeData(comments) } }
+            .subscribe {
+                it?.let { comments -> adapter.takeData(comments) }
```

Comment:
instead of `it?.let {...}` could we filter on the ViewModel the output?, so we can unit test that logic.
On the `Oncreate` methods for actvities/fragments if posible only have the subscription and no more logic unless related with UI

Reply:
Done